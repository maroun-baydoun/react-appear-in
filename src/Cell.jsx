import React, { Fragment, useCallback, useState } from "react";
import AppearIn from "react-appear-in";

const Cell = () => {
  const [demoStarted, setDemoStarted] = useState(false);
  const [duration, setDuration] = useState("5");

  const placeholder = useCallback(
    (time) => (
      <div
        className="cell__placeholder"
        style={{ animationDuration: `${time}ms` }}
      />
    ),
    []
  );

  const chooseDuration = (e) => setDuration(e.target.value);

  return (
    <div className="cell-container">
      <div className={`cell ${demoStarted ? "cell--started" : ""}`}>
        {!demoStarted ? (
          <Fragment>
            <div className="cell__options-container">
              <p>Make the component appear after: </p>
              <div className="">
                <input
                  type="radio"
                  value="5"
                  name="duration"
                  id="duration-5"
                  checked={duration === "5"}
                  onChange={chooseDuration}
                />
                <label htmlFor="duration-5">5s.</label>

                <input
                  type="radio"
                  value="10"
                  name="duration"
                  id="duration-10"
                  checked={duration === "10"}
                  onChange={chooseDuration}
                />
                <label htmlFor="duration-10">10s.</label>

                <input
                  type="radio"
                  value="15"
                  name="duration"
                  id="duration-15"
                  checked={duration === "15"}
                  onChange={chooseDuration}
                />
                <label htmlFor="duration-15">15s.</label>
              </div>
              <button className="button" onClick={() => setDemoStarted(true)}>
                Start
              </button>
            </div>
          </Fragment>
        ) : null}

        {demoStarted ? (
          <AppearIn
            seconds={Number.parseInt(duration, 10)}
            placeholder={placeholder}
          >
            <div className="cell__content">
              <div>
                I am a wonderful React component, and I appeared after{" "}
                {duration}s.
              </div>
            </div>
          </AppearIn>
        ) : null}
      </div>
    </div>
  );
};

export default Cell;
