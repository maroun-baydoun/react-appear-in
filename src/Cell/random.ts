
export const generateRandomNumber = (min: number, max: number) =>
  Math.floor(Math.random() * max) + min ;

export const generateRandomTime = () => ({
  milliseconds: generateRandomNumber(10, 100),
  minutes: generateRandomNumber(0, 1),
  seconds: generateRandomNumber(5, 30),
});
