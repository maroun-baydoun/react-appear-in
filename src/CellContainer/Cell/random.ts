
export const generateRandomNumber = (min: number, max: number) =>
  Math.floor(Math.random() * max) + min ;

export const generateRandomTime = () => ({
  milliseconds: generateRandomNumber(10, 100),
  minutes: generateRandomNumber(0, 2),
  seconds: generateRandomNumber(5, 30),
});
