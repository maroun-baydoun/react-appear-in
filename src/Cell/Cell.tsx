import React , { PureComponent } from 'react';
import AppearIn from 'react-appear-in';
import styled from 'styled-components';

import { generateRandomTime } from './random';

type State = {
  minutes: number,
  seconds: number,
  milliseconds: number,
};

class Cell extends PureComponent<{}, State> {
  state = { ...generateRandomTime() };

  render() {
    const { minutes, seconds, milliseconds } = this.state;
    return (
      <StyledContainer>
        <AppearIn minutes={minutes} seconds={seconds} milliseconds={milliseconds}>
          I appeared after {minutes}m. {seconds}s. {milliseconds}ms.
        </AppearIn>
      </StyledContainer>
    );
  }
}

const StyledContainer = styled('div')`
  background-color: darkgrey;
  color: #efefef;
  width: 150px;
  height: 100px;
  margin: 5px;
  padding: 5px;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export default Cell;
