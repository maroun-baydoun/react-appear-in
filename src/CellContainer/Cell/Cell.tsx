import React , { PureComponent } from 'react';
import AppearIn from 'react-appear-in';
import styled from 'styled-components';

import { generateRandomTime } from './random';

type State = {
  minutes: number,
  seconds: number,
  milliseconds: number,
};

class Cell extends PureComponent<{}, State> {
  state = { ...generateRandomTime() };

  render() {
    const { minutes, seconds, milliseconds } = this.state;
    return (
      <StyledContainer>
        <AppearIn minutes={minutes} seconds={seconds} milliseconds={milliseconds}>
          <StyledCellContent>
            <div>I appeared after</div>
            <TimeContainer>{minutes}m. {seconds}s. {milliseconds}ms.</TimeContainer>
          </StyledCellContent>
        </AppearIn>
      </StyledContainer>
    );
  }
}

const StyledCellContent = styled('div')`
  background-color: #efefef;
  border-radius: 10px;
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  justify-content: center;
  text-align: center;
`;

const StyledContainer = styled('div')`
  background-color: darkgrey;
  color: #ccc;
  width: 150px;
  height: 100px;
  margin: 5px;
  padding: 5px;
  border-radius: 10px;
  display: flex;
  flex-grow: 1;
  flex-direction: column;
`;

const TimeContainer = styled('div')`
  margin-top: 5px;
  color: #aaa;
`;

export default Cell;
