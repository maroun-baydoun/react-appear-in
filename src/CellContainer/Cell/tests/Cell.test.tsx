import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import Cell from '../Cell';

describe('<Cell/>', () => {
  const render = () => ReactTestRenderer.create(<Cell/>);

  it('renders component', () => {
    expect(render().toJSON()).toMatchSnapshot();
  });

  it('sets random state on mount', () => {
    const renderedComponent = render();

    // console.warn(renderedComponent.getInstance().state);
  });
});
