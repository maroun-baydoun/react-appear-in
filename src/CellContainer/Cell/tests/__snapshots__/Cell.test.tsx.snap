// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell/> renders component 1`] = `
<div
  className="sc-bwzfXH cWQFVt"
/>
`;
